echo/tracer.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/opentracing/opentracing-go/ext"
+	reporter "github.com/wavefronthq/wavefront-opentracing-sdk-go/reporter"
+	"github.com/wavefronthq/wavefront-opentracing-sdk-go/tracer"
+	wfTracer "github.com/wavefronthq/wavefront-opentracing-sdk-go/tracer"
+	wfApplication "github.com/wavefronthq/wavefront-sdk-go/application"
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+)
+
+// Tracer Global Tracer
+var Tracer opentracing.Tracer
+var globalTracerConfig *tracerConfig
+
+// InitTracer initialize tracer object, initialize the GlobalTracer with this newly created,
+// tracer object.
+// Call utils/wavefront.Tracer to access the tracer object
+//For proxy sender send token as empty string
+func InitTracer(cfg Config) error {
+
+	//Constructing globalTracerConfig from config file
+	tracerConfig, err := readConfigFromYaml(cfg.CfgFile)
+	if err != nil {
+		return fmt.Errorf("error reading config file: %w", err)
+	}
+	//Registering routes
+	tracerConfig, err = readConfigFromYaml(cfg.RoutesFile)
+	if err != nil {
+		return fmt.Errorf("error reading routes file: %w", err)
+	}
+
+	var sender senders.Sender
+	if cfg.DirectCfg != nil {
+		// Create the direct sender
+		sender, err = senders.NewDirectSender(cfg.DirectCfg)
+		if err != nil {
+			return err
+		}
+	} else if cfg.ProxyCfg != nil {
+		// Create the proxy sender
+		sender, err = senders.NewProxySender(cfg.ProxyCfg)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New(color.Error.Sprint("Wavefront Middleware... at least one of Direct Sender or Proxy Sender configuration must be passed"))
+	}
+
+	appTags := wfApplication.New(tracerConfig.Application, tracerConfig.Service)
+	if len(tracerConfig.Cluster) > 0 {
+		appTags.Cluster = tracerConfig.Cluster
+	}
+	if len(tracerConfig.Shard) > 0 {
+		appTags.Shard = tracerConfig.Shard
+	}
+	if tracerConfig.CustomApplicationTags.StaticTags != nil && len(tracerConfig.CustomApplicationTags.StaticTags) > 0 {
+		for k, v := range tracerConfig.CustomApplicationTags.StaticTags {
+			appTags.CustomTags[k] = v
+		}
+	}
+
+	wfReporter := reporter.New(sender, appTags, reporter.Source(tracerConfig.Source))
+
+	//Initialising tracer with RateSampler and DurationSampler if given in conifg
+	//If values for RateSampler and DurationSampler is not given, it initalises standard tracer with default values
+	var samplers []wfTracer.Option
+	if tracerConfig.DurationSampler > 0 {
+		samplers = append(samplers, wfTracer.WithSampler(tracer.DurationSampler{Duration: time.Duration(tracerConfig.DurationSampler) * time.Second}))
+	}
+	if tracerConfig.RateSampler > 0 {
+		samplers = append(samplers, wfTracer.WithSampler(tracer.RateSampler{Rate: tracerConfig.RateSampler}))
+	}
+	Tracer = wfTracer.New(wfReporter, samplers...)
+
+	//Instantiating Global Tracer
+	opentracing.SetGlobalTracer(Tracer)
+
+	//Enabling Middleware
+	cfg.EchoWeb.Use(TracingHandler)
+
+	log.Println(color.Success.Sprint("Wavefront Middleware... Tracer Initialized..."))
```

Comment:
similar comment. drop the color dep and log `wavefront middleware: tracer initialized`